src/formTimingTracking.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Form timing tracking
+ * 
+ * This script analyses the interactions with <form> fields.
+ */
+
+ export default (subscribe, {formNameAttribute, fieldNameAttribute}) => {
+    let fieldTimings = {};
+
+    const getFieldLabel = (field) => field.getAttribute(fieldNameAttribute);
+
+    const getFormName = (field) => field.closest("form").getAttribute(formNameAttribute);
+
+    const trackFormFieldEntry = (e) => {
+        let fieldName = getFieldLabel(e.target);
+        fieldTimings[fieldName] = new Date().getTime();
+    };
+    
+    const trackFormFieldLeave = (e) => {
+        let leaveType = e.type;
+        let formName = getFormName(e.target);
+        let fieldName = getFieldLabel(e.target);
+        // eslint-disable-next-line no-prototype-builtins
+        if (fieldTimings.hasOwnProperty(fieldName)) {
+            let timeSpent = new Date().getTime() - fieldTimings[fieldName];
+            if (timeSpent > 0 && timeSpent < 1800000) {
+                subscribe({ formName, fieldName, leaveType, timeSpent });
+            }
+            delete fieldTimings[fieldName];
+        }
+    };
+
+    document.querySelectorAll('input,select,textarea').forEach((elem) => {
```

Comment:
We don't need `button` and `datalist` elements?

Reply:
Button doesn't have time of being focused so it is not of our interest. As for datalist - I haven't ever considered such element, didn't even know about it to be honest. I haven't met such an element in any of customer environments, but sure, I added it.

Reply:
Just askin :) I don't know all your use cases.